src/sys/getrandom.rs

```diff
@@ -0,0 +1,48 @@
+//! Wrapper around getrandom.
+
+use crate::Result;
+use libc;
+
+libc_enum! {
+    /// How the random bytes should be filled
+    #[repr(u32)]
+    #[non_exhaustive]
+    pub enum RandomMode{
+        /// If  this  bit is set, then random bytes are drawn from the
+        /// random source (i.e., the same source as the /dev/random device)
+        ///  instead of the urandom source.
+        GRND_RANDOM,
+        /// By default, when reading from the random source, getrandom()
+        /// blocks if no random bytes are available, and when reading
+        /// from the urandom source, it blocks if the entropy pool has
+        /// not yet been initialized. If the GRND_NONBLOCK flag is
+        /// set, then getrandom() does not block in these cases
+        GRND_NONBLOCK,
+    }
+}
+
+/// Returns a vector of random bytes
+pub fn getrandom(size: usize, mode: RandomMode) -> Result<Vec<u8>> {
+    let mut buffer = vec![0; size];
```

Comment:
Rather than allocate internally, we should allow the user to pass in a buffer as a mutable slice.